config/build.ts

```diff
@@ -464,23 +458,55 @@ async function generatePlaylists(dir, processedSnippets: Dictionary<SnippetProce
         */
         const groupNumberRegex = /^(\d+\s)?(\w.*)$/;
 
-        items.forEach(item => {
+        let modifiedItems = cloneDeep(items);
+        modifiedItems.forEach(item => {
             item.group = item.group.replace(groupNumberRegex, '$2');
 
-            // Also remove "order", it's no longer needed (the snippets themselves are already in an ordered array in the YAML file)
+            // Also remove "order", "host" and "type", they are no longer needed (the snippets themselves are already in an ordered array in the YAML file, which is itself tied to host)
             delete item.order;
+            delete item.host;
+            delete item.type;
         });
 
-        let contents = jsyaml.safeDump(items, {
+        let contents = jsyaml.safeDump(modifiedItems, {
             skipInvalid: true /* skip "undefined" (e.g., for "order" on some of the snippets) */
         });
 
-        await writeFile(path.resolve(`${dir}/${host}.yaml`), contents);
+        /* Group private samples with public samples in one playlist */
+        let fileName = `playlists/${host}.yaml`;
+        await writeFile(path.resolve(fileName), contents);
+
+        status.complete(true /*success*/, creatingStatusText);
+    });
+
+    /* Creating view directory */
+    status.add(`Creating \'view\' folder`);
+    await rmRf('view');
+    await mkDir('view');
+    status.complete(true /*success*/, `Creating \'view\' folder`);
+
+    const allGroups = groupBy(
+        processedSnippets.values()
+            .filter((file) => !(file == null) && file.fileName !== 'default.yaml'),
+        'host');
+    let allPlaylistPromises = map(allGroups, async (items, host) => {
+        const creatingStatusText = `Creating ${host}.json`;
+        status.add(creatingStatusText);
+        items = sortBy(items, ['group', 'order', 'id']);
+
+        let hostMapping = {};
```

Comment:
Could you type it as a TypeScript dictionary?

    const hostMapping = {} as { [id: string]: string };